File: Adapter.js
//Interface do cliente:

class ITarget
{
    request(){}
}

class Client
{
    constructor(target){
        this.target = target;
    }

    makeRequest(){
        console.log("Fazendo uma requisição");
        this.target.request();
    }
}

class Adaptee
{
    sepecificRequest(){
        console.log("Requisição específica do Adaptee");
    }
}

//Adaptador:
class Adapter
{
    constructor(adaptee){
        this.adaptee = adaptee;
    }

    request(){
        this.adaptee.sepecificRequest();
    }
}

//Utilização do adaptador:
const adaptee = new Adaptee();
const adapter = new Adapter(adaptee);
const client = new Client(adapter);

client.makeRequest()

File: Adapter.py
#Interface do cliente:

class ITarget:

    def request(self):
        pass

class Client:

    def __init__(self, target):
        self.target = target
    

    def makeRequest(self):
        print("Fazendo uma requisição")
        self.target.request()
    

class Adaptee:

    def sepecificRequest(self):
        print("Requisição específica do Adaptee")
    

#Adaptador:
class Adapter(ITarget):

    def __init__(self, adaptee):
        self.adaptee = adaptee
    

    def request(self):
        self.adaptee.sepecificRequest()
    

#Utilização do adaptador:
adaptee = Adaptee()
adapter = Adapter(adaptee)
client = Client(adapter)

client.makeRequest();
